the-collab-lab/tcl-23-smart-shopping-list
# added firebase component for persistence

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description



Added a new firestore instance to the app with updated configuration information.  We also created a data storage instance for writing and reading information from our firestore database.

References:
[Firebase install docs](https://firebase.google.com/docs/web/setup)
## Related Issue

[Issue #1](https://github.com/the-collab-lab/tcl-23-smart-shopping-list/issues/2)

## Acceptance Criteria


## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
| ✓  | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before


Not Applicable -- non UI related feature



### After


<img width="1487" alt="Screen Shot 2021-04-09 at 6 04 10 PM" src="https://user-images.githubusercontent.com/23660204/114245257-1e582200-995e-11eb-8064-4a6cb50788d3.png">

Description
This PR adds two new components: one for adding new items to the database and another for displaying the list of items.

Display items from database
Inside that App component, we've used FirestoreCollection which is a hook that allows you to interact with a Firestore collection. Using this dependency, you can retrieve and monitor a collection value in Cloud Firestore.

This component will setup a listener and update whenever our collection is updated in Firestore.
More documentation on FirestoreCollection [here](https://github.com/csfrequency/react-firebase-hooks/tree/aeadd15b3f8ca642f52776edf2e4dad62a00d449/firestore)

Write items to the database
Our method is currently tested using the sendItem method, which sends a unique id when a button is clicked in the DOM.
We tested the GET Path by adding a document 'testDoc1' to our firestore database and observing this added document redering in the DOM. In our Collection React Fragment.





## Testing Steps / QA Criteria



1.  Start a local of our application using the npm start command. The app should now show a button and collection of data from Firestore.
2. Open the Chrome Dev tools  and select the console tab to observe changes
3. Click the Grocery List button to add documents to our collection (titled List)
4. Oberve new items successfully added via dev tools
5. Select the Firestore tab in the Firebase Dashboard
6. Open the list collection to see updated documents.
7. Manually add your own test document to test the GET method
8. Check the DOM again for new test document
